native/src/snappyclient/cpp/impl/ControlConnection.cpp

```diff
@@ -440,3 +449,35 @@ void ControlConnection::failoverExhausted(
   snappyEx.__set_serverInfo(failedServerString);
   throw snappyEx;
 }
+
+void ControlConnection::getConnectedHost(thrift::HostAddress& hostAddr,
+    thrift::HostAddress& connectedHost) {
+  boost::lock_guard<boost::mutex> controlConnLock(this->m_lock);
+
+  auto it = m_controlHostSet.find(hostAddr);
+  if (it != m_controlHostSet.end()){
+    connectedHost = it->second;
+    return;
+  }
+
+  for (auto iterator = m_controlHostSet.begin();
+      iterator != m_controlHostSet.end(); ++iterator) {
+    auto host = iterator->first;
+    if(host.__isset.isCurrent && host.isCurrent){
+      connectedHost = host;
+    }
+  }
```

Comment:
correct space formatting (select new code and use Ctrl-Shift-F in eclipse for example)

Reply:
Corrected formatting